testing/test_modal.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+from wait_for import wait_for
+
+from widgetastic.widget import View
+from widgetastic_patternfly import Modal, Button
+
+# Values from testing_page.html modal
+modal_id = 'myModal'
+title = 'Modal Title'
+
+
+def test_generic_modal(browser):
+    """
+    Test the modal, including all methods/properties
+
+    Test against modal defined in testing_page.html
+    :param browser: browser fixture
+    """
+    class TestView(View):
+        """ Dummy page matching testing_page.html elements"""
+        button = Button('Launch demo modal')
+        modal = Modal(id=modal_id)
+
+    view = TestView(browser)
+    assert not view.modal.is_displayed
+
+    # Open the modal
+    assert not view.button.disabled
+    view.button.click()
+    wait_for(lambda: view.modal.is_displayed, delay=0.5, num_sec=5)
+
+    assert view.modal.title == title
+
+    # close the modal via the "x"
+    view.modal.close()
+    view.flush_widget_cache()
+    wait_for(lambda: not view.modal.is_displayed, delay=0.5, num_sec=5)
+
+    # # open modal again
+    view.button.click()
+    wait_for(lambda: view.modal.is_displayed, delay=0.5, num_sec=5)
+    # make sure buttons are not disabled
+    assert not view.modal.footer.dismiss.disabled
+    assert not view.modal.footer.accept.disabled
+    # make sure the cancel button works
+    view.modal.dismiss()
+
+    wait_for(lambda: not view.modal.is_displayed, delay=0.1, num_sec=5)
```

Comment:
I guess you need to add at least one case when you accept modal

Reply:
For the reference markup, nothing happens when I click "Save", do we need need to modify the markup to `Save` on click?

Reply:
it would be great to modify it to fill some input field with any value and check it.